test/tests.py

```diff
@@ -100,28 +106,35 @@ def wait_for_cns(self, timeout=60):
                 if r.status_code == 200:
                     break
             except requests.exceptions.ConnectionError:
-                pass
-            timeout -= 1
-            time.sleep(1)
+                timeout -= 1
+                time.sleep(1)
         else:
             self.fail("nginx has not become reachable at {}"
                       .format(self.nginx_cns))
 
+    def get_possible_backend_ips(self, timeout=60):
+        _, ips = self.get_service_ips('backend')
+        ips.sort()
+        return ips
 
-    def compare_backends(self):
-        expected = self.get_service_instances_from_consul('backend').sort()
-        actual = list(set([self.query_for_backend() for _ in range(10)])).sort()
-        self.assertEqual(expected, actual,
-                         'Expected {} but got {} for Nginx backends'
-                         .format(expected, actual))
-
-    def query_for_backend(self):
-        r = requests.get('http://{}'.format(self.nginx_cns)) # TODO: SSL
-        if r.status_code != 200:
-            self.fail('Expected 200 OK but got {}'.format(r.status_code))
-        return r.text.strip('HelloWorld\n ')
-
+    def compare_backends(self, expected, timeout=60):
+        expected.sort()
+        patt = 'server \d{2,3}\.\d{2,3}\.\d{2,3}\.\d{2,3}\:3001;'
+        while timeout > 0:
+            conf = self.docker_exec('nginx_1',
```

Comment:
It should, but if `nginx_1` doesn't exist at this point, the test should fail anyways.